tests/integration/test_vectorizers.py

```diff
@@ -238,6 +240,29 @@ def bad_return_type(text: str) -> str:
         )
 
 
+def test_dtypes(vectorizer):
+    words = "hello"
+
+    raw = vectorizer.embed(words, as_buffer=False)
+
+    default = vectorizer.embed(words, as_buffer=True)
+    assert buffer_to_array(default, dtype="float32") == raw
+
+    float16 = vectorizer.embed(words, as_buffer=True, dtype="float16")
+    # assert buffer_to_array(float16, dtype="float16") == raw # fails
```

Comment:
For `float16` and `bfloat16` converting to a from this byte array results in rounding errors. Unclear how much of an issue this is, but it will fail on exact match checks.

Reply:
Why is this happening? Is it avoidable?

Reply:
Python's native dtype is float64, so when we convert to (b)float16 it's lossy. Converting back from bytes to Python float won't match exactly. It's not a bug, just something that we can keep in mind.

Reply:
Ahhh right. 👍🏼 makes complete sense. I misunderstood initially.